client/context/viper.go

```diff
@@ -15,13 +20,48 @@ func NewCoreContextFromViper() core.CoreContext {
 	if nodeURI != "" {
 		rpc = rpcclient.NewHTTP(nodeURI, "/websocket")
 	}
+	chainID := viper.GetString(client.FlagChainID)
+	// if chain ID is not specified manually, read chain ID from genesis file if present
+	if chainID == "" {
+		cfg, err := tcmd.ParseConfig()
+		if err == nil {
+			genesisFile := cfg.GenesisFile()
+			bz, err := ioutil.ReadFile(genesisFile)
+			if err == nil {
+				var doc tmtypes.GenesisDoc
+				err = json.Unmarshal(bz, &doc)
+				if err == nil {
+					chainID = doc.ChainID
+				}
+			}
+		}
+	}
 	return core.CoreContext{
-		ChainID:         viper.GetString(client.FlagChainID),
+		ChainID:         chainID,
 		Height:          viper.GetInt64(client.FlagHeight),
 		TrustNode:       viper.GetBool(client.FlagTrustNode),
 		FromAddressName: viper.GetString(client.FlagName),
 		NodeURI:         nodeURI,
 		Sequence:        viper.GetInt64(client.FlagSequence),
 		Client:          rpc,
+		Decoder:         nil,
+		AccountStore:    "main",
+	}
+}
+
+// Automatically set sequence number
+func AutoSequence(ctx core.CoreContext) (core.CoreContext, error) {
```

Comment:
This function name is slightly misleading because it's not always auto-sequencing it's only sequencing when there isn't a sequence number provided - but I get that you're trying to avoid dup code in all the command functions.. Maybe a more appropriate name would be `SetSequence` or `EnsureSequence`

Reply:
Changed to `EnsureSequence`.